packages/block-editor/src/components/link-control/use-search-handler.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * WordPress dependencies
+ */
+import { getProtocol, prependHTTP } from '@wordpress/url';
+import { useCallback } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * External dependencies
+ */
+import { startsWith } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import isURLLike from './is-url-like';
+import { CREATE_TYPE } from './constants';
+
+export const handleNoop = () => Promise.resolve( [] );
+
+export const handleDirectEntry = ( val ) => {
+	let type = 'URL';
+
+	const protocol = getProtocol( val ) || '';
+
+	if ( protocol.includes( 'mailto' ) ) {
+		type = 'mailto';
+	}
+
+	if ( protocol.includes( 'tel' ) ) {
+		type = 'tel';
+	}
+
+	if ( startsWith( val, '#' ) ) {
+		type = 'internal';
+	}
+
+	return Promise.resolve( [
+		{
+			id: val,
+			title: val,
+			url: type === 'URL' ? prependHTTP( val ) : val,
+			type,
+		},
+	] );
+};
+
+export const handleEntitySearch = async (
+	val,
+	args,
+	fetchSearchSuggestions,
+	directEntryHandler
+) => {
+	let results = await Promise.all( [
+		fetchSearchSuggestions( val, {
+			...( args.isInitialSuggestions ? { perPage: 3 } : {} ),
```

Comment:
Why wrap in Object and spread here?

Reply:
It's just an old code fragment moved over from `link-control/index.js`. It would be nice to polish rough edges like that but I didn't want to spend time on that in this PR.